core/CronArchive.php

```diff
@@ -1333,4 +1381,110 @@ private function makeWebsiteIdArchiveList(array $websitesIds)
+    {
+        if (empty($archivesToProcess)) {
+            return false;
+        }
+
+        return $archivesToProcess[0]['period'] != $period;
+    }
+
+    private function hasDifferentDoneFlagType(array $archivesToProcess, $name)
+    {
+        if (empty($archivesToProcess)) {
+            return false;
+        }
+
+        $existingDoneFlagType = $this->getDoneFlagType($archivesToProcess[0]['name']);
+        $newArchiveDoneFlagType = $this->getDoneFlagType($name);
+
+        return $existingDoneFlagType != $newArchiveDoneFlagType;
+    }
+
+    private function getDoneFlagType($name)
+    {
+        if ($name == 'done') {
+            return 'all';
+        } else {
+            return 'segment';
+        }
+    }
+
+    private function canSkipArchiveBecauseNoPoint(array $invalidatedArchive)
+    {
+        $site = new Site($invalidatedArchive['idsite']);
+
+        $periodLabel = $this->periodIdsToLabels[$invalidatedArchive['period']];
+        $dateStr = $periodLabel == 'range' ? ($invalidatedArchive['date1'] . ',' . $invalidatedArchive['date2']) : $invalidatedArchive['date1'];
+        $period = PeriodFactory::build($periodLabel, $dateStr);
+
+        $segment = new Segment($invalidatedArchive['segment'], [$invalidatedArchive['idsite']]);
+
+        $params = new Parameters($site, $period, $segment);
+
+        $loader = new Loader($params);
+        if ($loader->canSkipThisArchive()) { // if no point in archiving, skip
+            return true;
+        }
+
+        // if valid archive already exists, do not re-archive
+        $minDateTimeProcessedUTC = Date::now()->subSeconds(Rules::getPeriodArchiveTimeToLiveDefault($periodLabel));
+        $archiveIdAndVisits = ArchiveSelector::getArchiveIdAndVisits($params, $minDateTimeProcessedUTC, $includeInvalidated = false);
+
+        $idArchive = $archiveIdAndVisits[0];
+        return !empty($idArchive);
+    }
+
+    private function shouldSkipArchiveBecauseLowerPeriodOrSegmentIsInProgress(array $archiveToProcess)
+    {
+        $inProgressArchives = $this->cliMultiRequestParser->getInProgressArchivingCommands();
+
+        $periodLabel = $this->periodIdsToLabels[$archiveToProcess['period']];
+        $archiveToProcess['periodObj'] = PeriodFactory::build($periodLabel, $archiveToProcess['date1']);
+
+        foreach ($inProgressArchives as $archiveBeingProcessed) {
+            $archiveBeingProcessed['periodObj'] = PeriodFactory::build($archiveBeingProcessed['period'], $archiveBeingProcessed['date']);
+
+            if ($this->isArchiveOfLowerPeriod($archiveToProcess, $archiveBeingProcessed)) {
+                return "lower period in progress (period = {$archiveBeingProcessed['period']}, date = {$archiveBeingProcessed['date']})";
+            }
+
+            if ($this->isArchiveNonSegmentAndInProgressArchiveSegment($archiveToProcess, $archiveBeingProcessed)) {
+                return "segment archive in progress for same site/period ({$archiveBeingProcessed['segment']})";
```

Comment:
fyi segment might not be defined here 
![image](https://user-images.githubusercontent.com/273120/82970577-b4c07500-a024-11ea-84c7-c14afc6162d3.png)

got `Notice - Undefined index: segment`